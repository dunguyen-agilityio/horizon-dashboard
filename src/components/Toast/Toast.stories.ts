import type { Meta, StoryObj } from '@storybook/react';

import Toast from '.';
import { fn } from '@storybook/test';

const meta = {
  title: 'Components/Toast',
  component: Toast,
  tags: ['autodocs'],
} satisfies Meta<typeof Toast>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Success: Story = {
  args: {
    title: 'Success',
    message: 'This is messaging!',
    type: 'success',
    onClose: fn(),
  },
};

export const Error: Story = {
  args: { title: 'Error', message: 'This is messaging!', type: 'error' },
};

export const Info: Story = {
  args: { title: 'Info', message: 'This is messaging!', type: 'info' },
};

export const Warning: Story = {
  args: { title: 'Warning', message: 'This is messaging!', type: 'warning' },
};
