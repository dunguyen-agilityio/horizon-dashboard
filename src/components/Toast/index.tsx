'use client';

// types
import { ToastStatus } from '@/types/toast';

import { BoxIcon, Text } from '@/components';

// icons
import { Close } from '@/icons';

interface ToastProps {
  title: string;
  message: string;
  type?: ToastStatus;
  onClose?: () => void;
}

const typeStyles = {
  success: 'bg-success',
  error: 'bg-error',
  info: 'bg-secondary',
  warning: 'bg-warning',
};

const Toast = ({ title, message, type = 'info', onClose }: ToastProps) => (
  <div
    className={`fixed top-5 right-5 z-[1000] flex justify-between min-w-[300px] p-1
      rounded text-white text-lg transition-opacity duration-500 ease-in animate-slideDown ${typeStyles[type]}`}
  >
    <div className="flex flex-col justify-between min-h-12 px-3 py-1">
      <Text className="font-bold leading-none text-white" as="h2">
        {title}
      </Text>
      <Text className="font-normal text-sm !text-white leading-none opacity-9">
        {message}
      </Text>
    </div>
    <button
      onClick={onClose}
      className="w-5 h-5 hover:bg-white hover:bg-opacity-20 hover:rounded-full flex justify-center items-center"
    >
      <BoxIcon icon={<Close />} />
    </button>
  </div>
);

export default Toast;
