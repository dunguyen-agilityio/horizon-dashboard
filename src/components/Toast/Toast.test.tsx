import { render, screen } from '@testing-library/react';

import Toast from '.';

describe('Toast', () => {
  const mockOnClose = jest.fn();
  const ToastComponent = () =>
    render(
      <Toast
        message="Success message"
        type="success"
        onClose={mockOnClose}
        title="Success"
      />,
    );

  it('should match snapshot', () => {
    const { container } = ToastComponent();
    expect(container).toMatchSnapshot();
  });

  it('should renders the Toast component with the success type', () => {
    ToastComponent();

    expect(screen.getByText('Success message')).toBeInTheDocument();
  });

  it('should renders the Toast component with the error type', () => {
    render(
      <Toast
        message="Error message"
        type="error"
        onClose={mockOnClose}
        title="Success"
      />,
    );

    expect(screen.getByText('Error message')).toBeInTheDocument();
  });
});
