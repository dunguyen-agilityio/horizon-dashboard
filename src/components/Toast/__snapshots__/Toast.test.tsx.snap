// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should match snapshot 1`] = `
<div>
  <div
    class="fixed top-5 left-1/2 z-[1000] flex justify-between min-w-[300px] p-1
      rounded text-white text-lg transition-opacity duration-500 ease-in animate-slideDown bg-success"
  >
    <div
      class="flex flex-col justify-between gap-2 p-3"
    >
      <p
        class="text-primary dark:text-white text-medium leading-xl font-bold leading-none text-white"
      >
        Success
      </p>
      <p
        class="text-primary dark:text-white leading-xl font-normal text-sm text-amber-450 leading-none opacity-9"
      >
        Success message
      </p>
    </div>
    <i
      class="flex justify-center items-center hover:cursor-pointer w-7 h-7 hover:bg-white hover:bg-opacity-20 hover:rounded-full"
    >
      <svg
        class="w-full h-full"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M16 15.2931L20.6465 10.6466L21.3536 11.3537L16.7071 16.0002L21.3536 20.6466L20.6465 21.3537L16 16.7073L11.3536 21.3537L10.6465 20.6466L15.2929 16.0002L10.6465 11.3538L11.3536 10.6467L16 15.2931Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </i>
  </div>
</div>
`;
